### ISSUE TITLE
ToDo: diffs FF65-FF66

### ISSUE THREAD

**Issue by earthlng** [ghacksuserjs/ghacks-user.js#650]
FF66 is [scheduled](https://wiki.mozilla.org/Release_Management/Calendar) for release 19th Mar

:exclamation: When updating to FF66, backup your profile just in case (as you should anyway), because [the backend for extension storage is changing](https://blog.mozilla.org/addons/2019/02/15/extensions-in-firefox-66/)

[FF66 release notes](https://www.mozilla.org/en-US/firefox/66.0/releasenotes/) [when ready]
[FF66 for developers](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/66)
[FF66 compatibility](https://www.fxsitecompat.com/en-CA/versions/66/)

125 diffs ( 73 new, 29 gone, 23 different )

### new in v66.0:

* [x] pref("browser.newtabpage.activity-stream.feeds.discoverystreamfeed", true); - https://github.com/ghacksuserjs/ghacks-user.js/commit/2fcec590b41d21943a7dee8a57c0f575999715d9
* pref("dom.block_external_protocol_in_iframes", false); - [will be turned on in 67](https://www.fxsitecompat.com/en-CA/docs/2019/loading-of-external-protocol-url-in-iframe-is-now-blocked/)

### removed, renamed or hidden in v66.0:

`DONE` - https://github.com/ghacksuserjs/ghacks-user.js/commit/a349662f69d9cf4d9eb90352f589a8ea240be24e

* [x] errorReporter - [1509888](https://bugzilla.mozilla.org/show_bug.cgi?id=1509888)
   * pref("browser.chrome.errorReporter.enabled", false);
   * pref("browser.chrome.errorReporter.submitUrl", "**h**ttps://sentry.prod.mozaws.net/api/339/store/");
* [x] pref("network.allow-experiments", true); - [1415625](https://bugzilla.mozilla.org/show_bug.cgi?id=1415625)

### changed in v66.0:

* [x] pref("media.autoplay.default", 1); // prev: 0 `2030` - * add `[DEFAULT]` tag - https://github.com/ghacksuserjs/ghacks-user.js/commit/b1aa1f561926cf6e49c96699307f7329146c42d7 - https://github.com/ghacksuserjs/ghacks-user.js/commit/3c4b312cc7f765e05440e6f5f8a9840599d23567
  * default **not** changed, they gradually roll it out
* [x] pref("media.autoplay.enabled.user-gestures-needed", true); // prev: false - https://github.com/ghacksuserjs/ghacks-user.js/commit/2fcec590b41d21943a7dee8a57c0f575999715d9

---

* pref("breakpad.reportURL", "**h**ttps://crash-stats.mozilla.org/report/index/"); // prev: "**h**ttps://crash-stats.mozilla.com/report/index/" `0350`
* pref("browser.safebrowsing.downloads.remote.url", "**h**ttps://sb-ssl.google.com/safebrowsing/clientreport/download?key=%GOOGLE_SAFEBROWSING_API_KEY%"); // prev: "**h**ttps://sb-ssl.google.com/safebrowsing/clientreport/download?key=%GOOGLE_API_KEY%" `0414`
* pref("browser.safebrowsing.provider.google.updateURL", "**h**ttps://safebrowsing.google.com/safebrowsing/downloads?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2&key=%GOOGLE_SAFEBROWSING_API_KEY%"); // prev: "**h**ttps://safebrowsing.google.com/safebrowsing/downloads?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2&key=%GOOGLE_API_KEY%" `0413`
* pref("browser.safebrowsing.provider.google4.dataSharingURL", "**h**ttps://safebrowsing.googleapis.com/v4/threatHits?$ct=application/x-protobuf&key=%GOOGLE_SAFEBROWSING_API_KEY%&$httpMethod=POST"); // prev: "**h**ttps://safebrowsing.googleapis.com/v4/threatHits?$ct=application/x-protobuf&key=%GOOGLE_API_KEY%&$httpMethod=POST" `0417`
* pref("browser.safebrowsing.provider.google4.gethashURL", "**h**ttps://safebrowsing.googleapis.com/v4/fullHashes:find?$ct=application/x-protobuf&key=%GOOGLE_SAFEBROWSING_API_KEY%&$httpMethod=POST"); // prev: "**h**ttps://safebrowsing.googleapis.com/v4/fullHashes:find?$ct=application/x-protobuf&key=%GOOGLE_API_KEY%&$httpMethod=POST" `0413`
* pref("browser.safebrowsing.provider.google4.updateURL", "**h**ttps://safebrowsing.googleapis.com/v4/threatListUpdates:fetch?$ct=application/x-protobuf&key=%GOOGLE_SAFEBROWSING_API_KEY%&$httpMethod=POST"); // prev: "**h**ttps://safebrowsing.googleapis.com/v4/threatListUpdates:fetch?$ct=application/x-protobuf&key=%GOOGLE_API_KEY%&$httpMethod=POST" `0413`
* pref("geo.wifi.uri", "**h**ttps://www.googleapis.com/geolocation/v1/geolocate?key=%GOOGLE_LOCATION_SERVICE_API_KEY%"); // prev: "**h**ttps://www.googleapis.com/geolocation/v1/geolocate?key=%GOOGLE_API_KEY%" `0210`

---

### ignore

<details><summary>click me for details</summary><p>

==NEW
```js
pref("app.normandy.remotesettings.enabled", false);
pref("browser.newtabpage.activity-stream.discoverystream.config", "{\"api_key_pref\":\"extensions.pocket.oAuthConsumerKey\",\"enabled\":false,\"show_spocs\":true,\"layout_endpoint\":\"https://getpocket.com/v3/newtab/layout?version=1&consumer_key=$apiKey&layout_variant=basic\"}");
pref("browser.newtabpage.activity-stream.improvesearch.handoffToAwesomebar", false);
pref("browser.privatebrowsing.searchUI", true);
pref("browser.sanitizer.loglevel", "Warn");
pref("channelclassifier.allowlist_example", false);
pref("devtools.debugger.event-listener-breakpoints", "[]");
pref("devtools.debugger.event-listeners-visible", false);
pref("devtools.debugger.features.event-listeners-breakpoints", false);
pref("devtools.debugger.features.log-points", false);
pref("devtools.debugger.features.original-blackbox", true);
pref("devtools.debugger.features.windowless-workers", false);
pref("devtools.inspector.flexboxHighlighter.combine", false);
pref("devtools.inspector.new-rulesview.enabled", false);
pref("devtools.inspector.scrollable-badges.enabled", false);
pref("devtools.responsive.metaViewport.enabled", false);
pref("devtools.webconsole.input.editor", false);
pref("dom.delay.block_external_protocol_in_iframes", 10);
pref("dom.input_events.conform_to_level_1", true);
pref("dom.inputevent.inputtype.enabled", true);
pref("dom.media.autoplay.autoplay-policy-api", false);
pref("dom.timeout.defer_during_load", true);
pref("dom.timeout.max_idle_defer_ms", 10000);
pref("dom.vr.openvr.action_input", false);
pref("gfx.webrender.all", false);
pref("gfx.webrender.debug.gpu-cache", false);
pref("gfx.webrender.debug.picture-caching", false);
pref("gfx.webrender.debug.primitives", false);
pref("gfx.webrender.debug.texture-cache.disable-shrink", false);
pref("gfx.webrender.enabled", false);
pref("javascript.options.dynamicImport", false);
pref("javascript.options.unboxed_objects", false);
pref("javascript.options.wasm_verbose", false);
pref("layout.css.scroll-anchoring.enabled", true);
pref("layout.css.scroll-anchoring.highlight", false);
pref("layout.dynamic-reflow-roots.enabled", false);
pref("media.autoplay.blackList-override-default", true);
pref("network.process.enabled", false);
pref("pdfjs.disableOpenActionDestination", true);
pref("pdfjs.historyUpdateUrl", false);
pref("performance.adjust_to_machine", false);
pref("performance.low_end_machine", false);
pref("privacy.trackingprotection.cryptomining.enabled", false);
pref("privacy.trackingprotection.fingerprinting.enabled", false);
pref("security.certerrors.permanentOverride", true);
pref("security.remote_settings.intermediates.bucket", "security-state");
pref("security.remote_settings.intermediates.checked", 0);
pref("security.remote_settings.intermediates.collection", "intermediates");
pref("security.remote_settings.intermediates.enabled", false);
pref("security.remote_settings.intermediates.signer", "onecrl.content-signature.mozilla.org");
pref("services.sync.prefs.sync.browser.contentblocking.category", true);
pref("services.sync.prefs.sync.browser.contentblocking.introCount", true);
pref("services.sync.prefs.sync.browser.crashReports.unsubmittedCheck.autoSubmit2", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.feeds.section.highlights", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.feeds.section.topstories", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.feeds.snippets", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.feeds.topsites", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.section.highlights.includeBookmarks", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.section.highlights.includeDownloads", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.section.highlights.includePocket", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.section.highlights.includeVisited", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.section.highlights.rows", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.section.topstories.rows", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.showSearch", true);
pref("services.sync.prefs.sync.browser.newtabpage.activity-stream.topSitesRows", true);
pref("services.sync.prefs.sync.browser.taskbar.previews.enable", true);
pref("services.sync.prefs.sync.media.eme.enabled", true);
pref("urlclassifier.features.cryptomining.blacklistTables", "base-cryptomining-track-digest256");
pref("urlclassifier.features.cryptomining.whitelistTables", "mozstd-trackwhite-digest256");
pref("urlclassifier.features.fingerprinting.blacklistTables", "base-fingerprinting-track-digest256");
pref("urlclassifier.features.fingerprinting.whitelistTables", "mozstd-trackwhite-digest256");
```

==REMOVED or HIDDEN
```js
pref("app.update.download.backgroundInterval", 0);
pref("apz.allow_checkerboarding", true);
pref("browser.chrome.errorReporter.logLevel", "Error");
pref("browser.chrome.errorReporter.projectId", "339");
pref("browser.chrome.errorReporter.publicKey", "c709cb7a2c0b4f0882fcc84a5af161ec");
pref("browser.chrome.errorReporter.sampleRate", "0.001");
pref("devtools.debugger.features.origial-blackbox", false);
pref("devtools.webconsole.jsterm.reverse-search", true);
pref("dom.event.returnValue.enabled", false);
pref("dom.ipc.scheduler.chaoticScheduling", false);
pref("dom.ipc.scheduler.preemption", false);
pref("dom.ipc.scheduler.threadCount", 2);
pref("dom.ipc.scheduler.useMultipleQueues", true);
pref("dom.vr.process.enabled", false);
pref("intl.charset.fallback.utf8_for_file", false);
pref("layout.css.background-blend-mode.enabled", true);
pref("layout.css.box-decoration-break.enabled", true);
pref("layout.css.color-adjust.enabled", true);
pref("layout.css.image-orientation.enabled", true);
pref("layout.css.isolation.enabled", true);
pref("layout.css.mix-blend-mode.enabled", true);
pref("layout.css.overflow.moz-scrollbars.enabled", false);
pref("layout.css.scroll-behavior.property-enabled", true);
pref("media.autoplay.ask-permission", false);
pref("network.http.accept.default", "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8");
pref("urlclassifier.flashInfobarTable", "except-flashinfobar-digest256");
```

==CHANGED
```js
pref("app.update.staging.enabled", true); // prev: false
pref("browser.newtabpage.activity-stream.asrouter.providers.onboarding", "{\"id\":\"onboarding\",\"type\":\"local\",\"localProvider\":\"OnboardingMessageProvider\",\"enabled\":true,\"exclude\":[]}"); // prev: "{\"id\":\"onboarding\",\"type\":\"local\",\"localProvider\":\"OnboardingMessageProvider\",\"enabled\":true,\"exclude\":[\"RETURN_TO_AMO_1\"]}"
pref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts", true); // prev: false
pref("browser.safebrowsing.provider.mozilla.lists", "base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256,block-flash-digest256,except-flash-digest256,allow-flashallow-digest256,except-flashallow-digest256,block-flashsubdoc-digest256,except-flashsubdoc-digest256,ads-track-digest256,social-track-digest256,analytics-track-digest256,base-fingerprinting-track-digest256,content-fingerprinting-track-digest256,base-cryptomining-track-digest256,content-cryptomining-track-digest256,fanboyannoyance-ads-digest256,fanboysocial-ads-digest256,easylist-ads-digest256,easyprivacy-ads-digest256,adguard-ads-digest256"); // prev: "base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256,block-flash-digest256,except-flash-digest256,allow-flashallow-digest256,except-flashallow-digest256,block-flashsubdoc-digest256,except-flashsubdoc-digest256,except-flashinfobar-digest256,ads-track-digest256,social-track-digest256,analytics-track-digest256"
pref("browser.security.newcerterrorpage.enabled", true); // prev: false
pref("browser.security.newcerterrorpage.mitm.enabled", true); // prev: false
pref("dom.ipc.processCount", 8); // prev: 4
pref("dom.keyboardevent.keypress.set_keycode_and_charcode_to_same_value", true); // prev: false
pref("dom.window.event.enabled", true); // prev: false
pref("extensions.webextensions.ExtensionStorageIDB.enabled", true); // prev: false
pref("gfx.webrender.picture-caching", true); // prev: false
pref("html5.flushtimer.initialdelay", 16); // prev: 120
pref("html5.flushtimer.subsequentdelay", 16); // prev: 120
pref("network.http.send_window_size", 1024); // prev: 0
pref("urlclassifier.disallow_completions", "test-malware-simple,test-harmful-simple,test-phish-simple,test-unwanted-simple,test-track-simple,test-trackwhite-simple,test-block-simple,goog-downloadwhite-digest256,base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256,block-flash-digest256,except-flash-digest256,allow-flashallow-digest256,except-flashallow-digest256,block-flashsubdoc-digest256,except-flashsubdoc-digest256,goog-passwordwhite-proto,ads-track-digest256,social-track-digest256,analytics-track-digest256,base-fingerprinting-track-digest256,content-fingerprinting-track-digest256,base-cryptomining-track-digest256,content-cryptomining-track-digest256,fanboyannoyance-ads-digest256,fanboysocial-ads-digest256,easylist-ads-digest256,easyprivacy-ads-digest256,adguard-ads-digest256"); // prev: "test-malware-simple,test-harmful-simple,test-phish-simple,test-unwanted-simple,test-track-simple,test-trackwhite-simple,test-block-simple,goog-downloadwhite-digest256,base-track-digest256,mozstd-trackwhite-digest256,content-track-digest256,mozplugin-block-digest256,mozplugin2-block-digest256,block-flash-digest256,except-flash-digest256,allow-flashallow-digest256,except-flashallow-digest256,block-flashsubdoc-digest256,except-flashsubdoc-digest256,except-flashinfobar-digest256,goog-passwordwhite-proto,ads-track-digest256,social-track-digest256,analytics-track-digest256"
```

</p></details>

**Comment by earthlng** [comment]
reserved

**Comment by Atavic** [comment]
```security.remote_settings.intermediates.*```

See: https://wiki.mozilla.org/Firefox/RemoteSettings

**Comment by Thorin-Oakenpants** [comment]
@earthlng aee the :exclamation: in OP ... I was wondering, how does that affect extensions in PB Mode? Note: assuming there is no fallback in place from the developer: e.g gorhill & browser.storage.local - see `https://github.com/gorhill/uBlock/commit/0d369cda21bbce23a7376e0f7b2847a3c7a6d3d8`

I have 21 extensions, 15 of them have entries in `profile/browser-extension-data` with data in storage.js. I assume they will all lose their settings in PB Mode? (and/or some data, e.g https-everywhere@eff.org's storage.js is 1.69mb, VM's file contains my two scripts, etc)

I'm pushed for time, but I might install a simple extension in my FF66 test profile and see what happens. Just thought I might save some time by asking you

**Comment by Thorin-Oakenpants** [comment]
Just dropping this here for a while for some visibility

FYI: https://bugzilla.mozilla.org/show_bug.cgi?id=1407366 landed in Nightly behind extra prefs

**Comment by earthlng** [comment]
Wow, you already narrowed it down a lot! Thanks Pants :kiss:

**Comment by Thorin-Oakenpants** [comment]
14 days ago .. waiting on you

**Comment by earthlng** [comment]
- `channelclassifier.allowlist_example` - safebrowsing related + is just a test pref anyway. [1511436 ](https://bugzilla.mozilla.org/show_bug.cgi?id=1511436) - safe to ignore
- `dom.media.autoplay.autoplay-policy-api` - still default false and we can ignore it for now
- `media.autoplay.blackList-override-default` - apparently makes sure site permissions take precedent over global autoplay allow/deny setting. Default true is fine and no need to add this to the user.js IMO.
- `network.process.enabled` - still default false atm. ignore for now
- `pdfjs.*` - not my problem
- `security.certerrors.permanentOverride` - questionable but an argument can be made for and against it. probably fine to go with the default and ignore
- `extensions.webextensions.ExtensionStorageIDB.enabled` - nothing to do with privacy/security. ignore
- `media.autoplay.enabled.user-gestures-needed` - I guess this combined with `media.autoplay.default=1` disables autoplay until a user interacts with a site

**Comment by bogachenko** [comment]
ss
```js
pref("browser.security.newcerterrorpage.enabled", true); //false
pref("geo.wifi.uri", "https://www.googleapis.com/geolocation/v1/geolocate?key=%GOOGLE_LOCATION_SERVICE_API_KEY%"); // https://www.googleapis.com/geolocation/v1/geolocate?key=%GOOGLE_API_KEY%
pref("media.autoplay.default", 0); // 0=Allowed, 1=Blocked, 2=Prompt // nothing change ~ it is 1.
pref("media.navigator.mediadatadecoder_h264_enabled", true); // false
```

**Comment by earthlng** [comment]
66.0 changes since 66.0b7

#### new
```js
pref("browser.newtabpage.activity-stream.discoverystream.config", "{\"api_key_pref\":\"extensions.pocket.oAuthConsumerKey\",\"enabled\":false,\"show_spocs\":true,\"layout_endpoint\":\"https://getpocket.com/v3/newtab/layout?version=1&consumer_key=$apiKey&layout_variant=basic\"}"); // v66.0b7: {\"enabled\":false,\"show_spocs\":true,\"layout_endpoint\":\"https://getpocket.com/v3/newtab/layout?version=1&consumer_key=40249-e88c401e1b1f2242d9e441c4&layout_variant=basic\"}
pref("browser.privatebrowsing.searchUI", true); // false in v66.0b7
pref("javascript.options.unboxed_objects", false);
pref("layout.dynamic-reflow-roots.enabled", false); // true in v66.0b7
```

#### changed
~~pref("app.update.channel", "beta"); // prev: "release"~~
~~pref("app.update.url.details", "https://www.mozilla.org/%LOCALE%/firefox/beta/notes"); // prev: "https://www.mozilla.org/%LOCALE%/firefox/notes"~~
~~pref("app.update.url.manual", "https://www.mozilla.org/firefox/beta"); // prev: "https://www.mozilla.org/firefox/"~~
```js
pref("browser.newtabpage.activity-stream.asrouter.providers.onboarding", "{\"id\":\"onboarding\",\"type\":\"local\",\"localProvider\":\"OnboardingMessageProvider\",\"enabled\":true,\"exclude\":[]}"); // prev: "{\"id\":\"onboarding\",\"type\":\"local\",\"localProvider\":\"OnboardingMessageProvider\",\"enabled\":true,\"exclude\":[\"RETURN_TO_AMO_1\"]}"
pref("browser.safebrowsing.downloads.remote.url", "https://sb-ssl.google.com/safebrowsing/clientreport/download?key=%GOOGLE_SAFEBROWSING_API_KEY%"); // prev: "https://sb-ssl.google.com/safebrowsing/clientreport/download?key=%GOOGLE_API_KEY%"
pref("browser.safebrowsing.provider.google.updateURL", "https://safebrowsing.google.com/safebrowsing/downloads?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2&key=%GOOGLE_SAFEBROWSING_API_KEY%"); // prev: "https://safebrowsing.google.com/safebrowsing/downloads?client=SAFEBROWSING_ID&appver=%MAJOR_VERSION%&pver=2.2&key=%GOOGLE_API_KEY%"
pref("browser.safebrowsing.provider.google4.dataSharingURL", "https://safebrowsing.googleapis.com/v4/threatHits?$ct=application/x-protobuf&key=%GOOGLE_SAFEBROWSING_API_KEY%&$httpMethod=POST"); // prev: "https://safebrowsing.googleapis.com/v4/threatHits?$ct=application/x-protobuf&key=%GOOGLE_API_KEY%&$httpMethod=POST"
pref("browser.safebrowsing.provider.google4.gethashURL", "https://safebrowsing.googleapis.com/v4/fullHashes:find?$ct=application/x-protobuf&key=%GOOGLE_SAFEBROWSING_API_KEY%&$httpMethod=POST"); // prev: "https://safebrowsing.googleapis.com/v4/fullHashes:find?$ct=application/x-protobuf&key=%GOOGLE_API_KEY%&$httpMethod=POST"
pref("browser.safebrowsing.provider.google4.updateURL", "https://safebrowsing.googleapis.com/v4/threatListUpdates:fetch?$ct=application/x-protobuf&key=%GOOGLE_SAFEBROWSING_API_KEY%&$httpMethod=POST"); // prev: "https://safebrowsing.googleapis.com/v4/threatListUpdates:fetch?$ct=application/x-protobuf&key=%GOOGLE_API_KEY%&$httpMethod=POST"
```
pref("browser.security.newcerterrorpage.enabled", true); // prev: false
~~pref("extensions.webcompat-reporter.enabled", true); // prev: false~~
pref("geo.wifi.uri", "https://www.googleapis.com/geolocation/v1/geolocate?key=%GOOGLE_LOCATION_SERVICE_API_KEY%"); // prev: "https://www.googleapis.com/geolocation/v1/geolocate?key=%GOOGLE_API_KEY%"
~~pref("media.autoplay.default", 1); // prev: 0~~
~~pref("toolkit.telemetry.enabled", true); // prev: false~~

**Comment by earthlng** [comment]
moved `javascript.options.unboxed_objects` from new to ignore and `browser.newtabpage.activity-stream.asrouter.providers.onboarding` + `browser.security.newcerterrorpage.enabled` from changed to ignore. 

`browser.newtabpage.activity-stream.feeds.discoverystreamfeed` [handles new pocket ui for the new tab page](https://dxr.mozilla.org/mozilla-central/source/browser/components/newtab/lib/ActivityStream.jsm#367)

**Comment by Thorin-Oakenpants** [comment]
bump... wanna do something with these?
- `browser.newtabpage.activity-stream.feeds.discoverystreamfeed` - what exactly does it do. If we already disable pocket being shown, then this `new UI change` is a nothing-burger. Right? So we can ignore it?
- `media.autoplay.enabled.user-gestures-needed` - how do we word this. Can you find a bugzilla for me to read?
- I'm so tired
- I miss :cat2: .. do you think Batman finally got him?

**Comment by bogachenko** [comment]
@Thorin-Oakenpants 

> media.autoplay.enabled.user-gestures-needed - how do we word this. Can you find a bugzilla for me to read?

IDK look this:

http://forums.mozillazine.org/viewtopic.php?p=14813334
>according to Sören Hentzschel media.autoplay.enabled is deprecated and no longer available

bugzilal
https://bugzilla.mozilla.org/show_bug.cgi?id=1420389 !
https://bugzilla.mozilla.org/show_bug.cgi?id=1427966
https://bugzilla.mozilla.org/show_bug.cgi?id=1407113

perhaps the earth will find better.

**Comment by Thorin-Oakenpants** [comment]
Yeah, sorry - that was all directed at the Big E, because I'm tired and can't be F'ed right now. I didn't search myself, and I know what the pref does, kind of (from another issue here at this repo), but wanted to read what the devs said it is meant to do: i.e I want to get the wording right when I draft something

> according to Sören Hentzschel media.autoplay.enabled is deprecated and no longer available

yes, we already have that marked as removed in FF63+ in the user.js along with the relevant bugzilla where it was deprecated

**Comment by bogachenko** [comment]
:disappointed:

**Comment by earthlng** [comment]
`browser.newtabpage.activity-stream.feeds.discoverystreamfeed` - FF67 will have a bunch more of these discoverystreamfeed prefs. I'd say let's just kill this shit now and add this pref to `0105c`

> - `media.autoplay.enabled.user-gestures-needed` - how do we word this

IDK. 2 things to note though: setting it to false [can apparently completely break playing videos on certain sites](https://bugzilla.mozilla.org/show_bug.cgi?id=1376321#c12), and setting it to false also blocks autoplay of muted videos (and audio?).
The former can probably be worked around by using a site permission and the latter is IMO a good thing anyway because it stops sites from abusing the soon-to-be default config by starting a video muted and then changing the volume a second later or so.

**Comment by earthlng** [comment]
Thanks Pants for doing this one pretty much all by yourself! :kiss:
